.github/api_best_practices.md

```diff
@@ -0,0 +1,198 @@
+
+The noun portion of methods that act on a single resource **must** be singular e.g. `GetDeployment`. Those methods that 
+act on the collection of resources **must** be plural e.g. `ListDeployments`, `DeleteDeployments`. Avoid prepositions 
+(e.g. for, by) in method names as much as possible. Typically, this can be addressed by using a distinct verb, 
+adding a field to the request message, or restructuring _VerbNoun_.
+
+<table>
+<tr>
+<td><b>Instead of</b></td><td><b>Use</b></td>
+</tr>
+<tr>
+<td>
+
+`GetBaselineGeneratedNetworkPolicyForDeployment`
+
+</td>
+
+<td>
+
+```
+GenerateDeploymentNetworkPolicy
+
+GenerateDeploymentNetworkPolicyRequest {
+  bool from_baseline;  
+  bool from_network_flows;
+}
+```
+
+</td>
+</tr>
+
+<tr>
+<td>
+
+`RunPolicyScanForDeployment`  
+
+</td>
+<td>
+
+`RunDeploymentPolicyScan`
+
+</td>
+</tr>
+
+<tr>
+
+<td>
+
+`DeleteDeploymentsByQuery`
+
+</td>
+<td>
+
+```
+DeleteDeployments
+
+DeleteDeploymentsRequest {
+  string query; 
+}
+```
+
+</td>
+</tr>
+</table>
+
+### Message Name
+The request and response messages **must** be named after method names with suffix `Request` and `Response` unless 
+the request/response type is an empty message. Generally, resource type as response message should be avoided 
+e.g. use `GetDeploymentResponse` response instead of `Deployment`. This allows augmenting the response with 
+supplemental information in the future.
+
+| Verb                      | Noun           | Method name         | Request message            | Response message            |
+|---------------------------|----------------|---------------------|----------------------------|-----------------------------|
+| List                      | Deployment     | `ListDeployments`   | `ListDeploymentRequest`    | `ListDeploymentResponse`    |
+| Get                       | Deployment     | `GetDeployment`     | `GetDeploymentRequest`     | `GetDeploymentResponse`     |   
+| Update                    | Deployment     | `UpdateDeployment`  | `UpdateDeploymentRequest`  | `UpdateDeploymentResponse`  |  
+| Delete                    | Deployment     | `DeleteDeployment`  | `DeleteDeploymentRequest`  | `google.protobuf.Empty`     |     
+| Get                       | ReportStatus   | `GetReportStatus`   | `GetReportStatusRequest`   | `GetReportStatusResponse`   |  
+| Run                       | ComplianceScan | `RunComplianceScan` | `RunComplianceScanRequest` | `RunComplianceScanResponse` | 
+
+Avoid prepositions as much as possible (e.g. “for”, “with”; `DeploymentWithProcessInfo`, `DeploymentWithImageScan`).
+In case such a need arises, add a field to the request message and response message.
+
+<table>
+<tr>
+<td><b>Instead of</b></td><td><b>Use</b></td>
+</tr>
+<tr>
+<td>
+
+`GetDeploymentWithImageScanRequest`
+
+</td>
+<td>
+
+```
+GetDeploymentRequest {
+  bool with_image_scan;
+}
+```
+
+</td>
+</tr>
+<tr>
+<td>
+
+`GetDeploymentWithImageScanResponse`
+
+</td>
+<td>
+
+```
+GetDeploymentWithImageScanResponse
```

Comment:
Is there no mistake to have `GetDeploymentWithImageScanResponse` second time here?